### PR TITLE
Feature/tasks

### PR DESCRIPTION
**Pull request by DaniilKarunou**
## Overview

This pull request includes a series of updates and improvements related to the Task management functionality in the application. The enhancements cover various aspects such as entity definitions, validation, service layer improvements, and REST API integrations. The changes aim to refine task handling, ensure data integrity, and improve error handling across the application.

### Entity Definition

- Created `Task` entity with fields: `id`, `title`, `description`, `priority`, `deadline`, and `status`.
- Applied Lombok annotations to reduce boilerplate code and manage getters, setters, constructors, and other methods efficiently.

### DTO and Mapper Integration

- Introduced `TaskRequest` as a DTO for incoming task data.
- Added `TaskResponse` DTO for responses (optional, but can be omitted if only `Task` is used directly).
- Implemented `TaskMapper` using MapStruct to map between `Task` and `TaskRequest` entities.

### Service Layer

- Implemented validation for task data, ensuring deadlines are in the future and titles are not empty.
- Added `TaskService` methods to handle task creation, updating, retrieval, and deletion.
- Integrated validation using Jakarta Validation API and Hibernate Validator.
- Included custom validation logic to check for data integrity and consistency.

### Controller Implementation

- Created `TaskController` with endpoints for CRUD operations:
  - `GET /api/tasks` - Retrieve all tasks.
  - `GET /api/tasks/{id}` - Retrieve a specific task by ID.
  - `POST /api/tasks` - Create a new task.
  - `PUT /api/tasks/{id}` - Update an existing task by ID.
  - `DELETE /api/tasks/{id}` - Delete a task by ID.
- Used `ResponseEntity` to handle HTTP responses and status codes.
- Implemented exception handling within the controller to catch and respond to common errors.

### Exception Handling

- Added a global exception handler to manage common exceptions, including `NotFoundException` and validation errors.
- Utilized `@ControllerAdvice` and `@ExceptionHandler` annotations to provide consistent error responses across the application.

### Maven Configuration

- Updated `pom.xml` to include necessary dependencies for MapStruct, Lombok, and Jakarta Validation.
- Configured `maven-compiler-plugin` to handle annotation processing for MapStruct.

### CODE REVIEW

File: src/main/java/com/todoapp/dto/TaskRequest.java
[the remark this answers is not in the document]
Comment (DaniilKarunou) [reply src/main/java/com/todoapp/dto/TaskRequest.java:1]: Fixed

File: src/main/java/com/todoapp/entity/task/Task.java
[the remark this answers is not in the document]
Comment (DaniilKarunou) [reply src/main/java/com/todoapp/entity/task/Task.java:1]: Fixed

File: src/main/java/com/todoapp/service/TaskService.java
[the remark this answers is not in the document]
Comment (DaniilKarunou) [reply src/main/java/com/todoapp/service/TaskService.java:1]: Fixed

**Review by KarinaYarmosh — COMMENTED** [review]
goog ga(me)y

**Review by KarinaYarmosh — APPROVED** [review]

**Pull request merged**